FluentUI/src/main/java/com/microsoft/fluentui/contextualcommandbar/ContextualCommandBar.kt

```diff
@@ -0,0 +1,262 @@
+        defStyleAttr: Int = 0
+) : FrameLayout(context, attrs, defStyleAttr) {
+
+    private var horizontalScrollView = HorizontalScrollView(context)
+    private var commandContainer = LinearLayout(context)
+
+    // Layout configuration
+    private var groupSpace = 0
+    private var itemSpace = 0
+    private var itemPaddingVertical = 0
+    private var itemPaddingHorizontal = 0
+
+    // Dismiss button configuration
+    var showDismiss = false
+        set(value) {
+            field = value
+            setDismissButtonVisible(value)
+        }
+
+    @DrawableRes
+    var dismissIcon: Int = 0
+        set(value) {
+            if (value == 0) {
+                return
+            }
+
+            field = value
+            dismissButton?.setImageResource(dismissIcon)
+        }
+    private var dismissButton: ImageView? = null
+    private var dismissButtonContainer: LinearLayout? = null
+
+    var dismissListener: (() -> Unit)? = null
+    var itemClickListener: OnItemClickListener? = null
+
+    init {
+        removeAllViews()
+        horizontalScrollView.removeAllViews()
+        addView(horizontalScrollView)
+        horizontalScrollView.addView(commandContainer)
+
+        with(commandContainer) {
+            orientation = LinearLayout.HORIZONTAL
+            gravity = Gravity.CENTER
+        }
+
+        with(horizontalScrollView) {
+            isVerticalScrollBarEnabled = false
+            isHorizontalScrollBarEnabled = false
+        }
+
+        attrs?.let {
+            val styledAttributes = context.theme.obtainStyledAttributes(
+                    it,
+                    R.styleable.ContextualCommandBar2,
+                    0,
+                    0
+            )
+
+            try {
+                groupSpace = styledAttributes.getDimensionPixelSize(
+                        R.styleable.ContextualCommandBar2_groupSpace,
+                        resources.getDimensionPixelSize(R.dimen.fluentui_contextual_command_bar_default_group_space)
+                )
+                itemSpace = styledAttributes.getDimensionPixelSize(
+                        R.styleable.ContextualCommandBar2_itemSpace,
+                        resources.getDimensionPixelSize(R.dimen.fluentui_contextual_command_bar_default_item_space)
+                )
+                showDismiss = styledAttributes.getBoolean(
+                        R.styleable.ContextualCommandBar2_showDismiss,
+                        false
+                )
+            } finally {
+                styledAttributes.recycle()
+            }
+
+            itemPaddingVertical = resources.getDimensionPixelSize(
+                    R.dimen.fluentui_contextual_command_bar_default_item_padding_vertical
+            )
+            itemPaddingHorizontal = resources.getDimensionPixelSize(
+                    R.dimen.fluentui_contextual_command_bar_default_item_padding_horizontal
+            )
+        }
+
+        // Initialize dismiss button
+        addDismissButton()
```

Comment:
is this needed to be added if visibility is off. Should we not add it only if showVisible is true ?